app/models/backend_api.rb

```diff
@@ -37,25 +37,20 @@ class BackendApi < ApplicationRecord
 
   has_system_name(uniqueness_scope: [:account_id])
 
-  scope :orphans, -> { where.has { id.not_in(BackendApiConfig.selecting { :backend_api_id }) } }
+  scope :orphans, -> {
+    where.has do
+      not_exists(BackendApiConfig.except(:order).by_backend_api(BabySqueel[:backend_apis].id).select(:id))
+    end
+  }
 
   scope :not_used_by, ->(service_id) {
-    # TODO: Baby Squeel
-    # It should be:
-    # where.has do
-    #   not_exists BackendApiConfig.by_service(service_id).by_backend_api(BabySqueel[:backend_apis].id).select(:id)
-    # end
-    # And that works for MySQL and Postgres but not Oracle
-    sql_query = <<~SQL
-      (
-        NOT EXISTS (
-          SELECT id
-          FROM backend_api_configs
-          WHERE service_id = ? AND backend_api_configs.backend_api_id = backend_apis.id
-        )
+    where.has do
+      not_exists(
+        BackendApiConfig.except(:order).select(:id)
+          .by_service(service_id)
+          .by_backend_api(BabySqueel[:backend_apis].id)
       )
-    SQL
-    where(sql_query, service_id)
+    end
   }
```

Comment:
What generates for Oracle now:
```sql
SELECT "BACKEND_APIS".*
FROM "BACKEND_APIS"
WHERE (
  NOT EXISTS(
    SELECT "BACKEND_API_CONFIGS"."ID"
    FROM "BACKEND_API_CONFIGS"
    WHERE "BACKEND_API_CONFIGS"."SERVICE_ID" = theServiceID
      AND "BACKEND_API_CONFIGS"."BACKEND_API_ID" = "BACKEND_APIS"."ID"
  )
)
```

Reply:
![image](https://user-images.githubusercontent.com/11318903/71095877-a2a4f580-21ad-11ea-8020-6643ca680e1d.png)